sdk/compiler/damlc/lib/DA/Cli/Damlc.hs

```diff
@@ -1717,3 +1761,18 @@ cmdUseDamlYamlArgs = \case
 withProjectRoot' :: ProjectOpts -> ((FilePath -> IO FilePath) -> IO a) -> IO a
 withProjectRoot' ProjectOpts{..} act =
     withProjectRoot projectRoot projectCheck (const act)
+
+addResolutionData :: Options -> IO Options
```

Comment:
Needing to add `addResolutionData` everywhere seems pretty tedious, but I see why we can't resort to IO in the parsing stage. Can we make `optionsParser` return a `Parser (IO Options)` that you then need to bind out of?

Reply:
We can have two variants - `OptionsF IO` and `OptionsF VoidF`. I remember toying with this and also noticed that it's messy, but the current approach definitely has limitations.

For example, I think you run `addResolutionData` to opts twice in `execDocTest`